### PR TITLE
Fix increment calculation in IncrementingPollingCounter

### PR DESCRIPTION
**Pull request by sywhang**
Fixes: #23504 

Found this while testing dotnet-counters tool - `IncrementingPollingCounter` should be reporting the "increment" between each period it gets reported. If `increment` is added with the function that collects it, it reports whatever `getCountFunction()` is returning and that is not what we want in the payload (this is what `PollingCounter` is for). Instead, it should be setting `increment` to whatever `getCountFunction` returned, so that the diff compared to the previously reported value can be shown in the payload. 

Adding test for this API as well.

### CODE REVIEW

**Comment by sywhang** [comment]
@dotnet-bot test this please

**Comment by sywhang** [comment]
CI failures are due to the tests being disabled in https://github.com/dotnet/coreclr/pull/23535/files

File: tests/src/tracing/eventcounter/incrementingpollingcounter.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                            Console.WriteLine($"Incorrect increment: {increment}.");
+                            Failed = true;
+                        }
+                    }
+                }
+            }
+        }
+
+
+        public static int failureCountCalled = 0;
```
Comment (noahfalk) [inline tests/src/tracing/eventcounter/incrementingpollingcounter.cs]: Can we change this from 'failure' to some other term ( ie _mockedCountCalled)? When "failure" shows up in a test case and it has a non-zero value, the kneejerk interpretation is that the test failed. In this case 'failure' appears to be a stand-in for 'some arbitrary thing we are pretending to count'

Comment (sywhang) [reply tests/src/tracing/eventcounter/incrementingpollingcounter.cs]: Ah, didn't consider that part. I'll follow up with another PR to fix that.

**Pull request merged**